src/include/catalog/catalog_accessor.h

```diff
@@ -355,6 +358,21 @@ class EXPORT CatalogAccessor {
    */
   common::ManagedPointer<execution::functions::FunctionContext> GetFunctionContext(proc_oid_t proc_oid);
 
+  /**
+   * Gets the statistics of a column from pg_statistic
+   * @param table_oid table oid of table
+   * @param col_oid column oid of column
+   * @return column statistics
+   */
+  std::unique_ptr<optimizer::ColumnStatsBase> GetColumnStatistics(table_oid_t table_oid, col_oid_t col_oid);
+
+  /**
+   * Gets the statistics of a table from pg_statistic
+   * @param table_oid table oid of table
+   * @return table statistics
+   */
+  optimizer::TableStats GetTableStatistics(table_oid_t table_oid);
```

Comment:
This is due to my lack of experience with C++ templates and polymorphism. I didn't want to unnecessarily allocate extra heap memory so I made this function return a raw object. However, it's my understanding that since `ColumnStatsBase` is an abstract class, I need to return a pointer to it and I can't return the class directly. If you have any suggestions on how to improve this please let me know.

Reply:
I guess let's figure out whether we want to return a copy or a reference. If we need to return a copy, then I think this is fine.

Reply:
Even if we don't return a copy, I still don't know how to get around using a pointer for `ColumnStatsBase` since it's an abstract class.